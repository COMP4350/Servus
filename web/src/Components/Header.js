import React from 'react';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import Button from '@material-ui/core/Button';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import { withRouter } from 'react-router-dom';

const useStyles = makeStyles(theme => ({
    root: {
        flexGrow: 1,
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },
    title: {
        [theme.breakpoints.down('xs')]: {
            flexGrow: 1,
        },
    },
    headerOptions: {
        display: 'flex',
        flex: 1,
        justifyContent: 'space-evenly',
    },
}));

const Header = props => {
    const { history } = props;
    const classes = useStyles();
    const [anchorEl, setAnchorEl] = React.useState(null);
    const open = Boolean(anchorEl);
    const theme = useTheme();
    const isMobile = useMediaQuery(theme.breakpoints.down('xs'));

    const handleMenu = event => {
        setAnchorEl(event.currentTarget);
    };

    const handleMenuClick = pageURL => {
        history.push(pageURL);
        setAnchorEl(null);
    };

    const handleButtonClick = pageURL => {
        history.push(pageURL);
    };

    const menuItems = [
        {
            menuTitle: 'Home',
            pageURL: '/',
        },
        {
            menuTitle: 'Contact',
            pageURL: '/contact',
        },
        {
            menuTitle: 'About',
            pageURL: '/about',
        },
        {
            menuTitle: 'Appointment',
            pageURL: '/appointment',
        },
    ];

    return (
        <div className={classes.root}>
            <AppBar position="static">
                <Toolbar>
                    <Typography variant="h6" className={classes.title}>
                        Servus
                    </Typography>
                    {isMobile ? (
                        <>
                            <IconButton
                                edge="start"
                                className={classes.menuButton}
                                color="inherit"
                                aria-label="menu"
                                onClick={handleMenu}>
                                <MenuIcon />
                            </IconButton>
                            <Menu
                                id="menu-appbar"
                                anchorEl={anchorEl}
                                anchorOrigin={{
                                    vertical: 'top',
                                    horizontal: 'right',
                                }}
                                keepMounted
                                transformOrigin={{
                                    vertical: 'top',
                                    horizontal: 'right',
                                }}
                                open={open}
                                onClose={() => setAnchorEl(null)}>
                                {menuItems.map(menuItem => {
                                    const { menuTitle, pageURL } = menuItem;
                                    return (
                                        <MenuItem
                                            onClick={() =>
                                                handleMenuClick(pageURL)
                                            }>
                                            {menuTitle}
                                        </MenuItem>
                                    );
                                })}
                            </Menu>
                        </>
                    ) : (
                        <div className={classes.headerOptions}>
                            <Button
                                variant="contained"
                                onClick={() => handleButtonClick('/')}>
                                HOME
                            </Button>
                            <Button
                                variant="contained"
                                onClick={() =>
                                    handleButtonClick('/appointment')
                                }>
                                APPOINTMENT
                            </Button>
                            <Button
                                variant="contained"
                                onClick={() => handleButtonClick('/contact')}>
                                CONTACT
                            </Button>
                            <Button
                                variant="contained"
                                onClick={() => handleButtonClick('/about')}>
                                ABOUT
                            </Button>
                        </div>
                    )}
                </Toolbar>
            </AppBar>
        </div>
    );
};

export default withRouter(Header);
