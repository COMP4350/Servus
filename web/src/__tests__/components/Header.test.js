import React from 'react';
import { render } from '@testing-library/react';
import Header from '../../components/Header';
import { BrowserRouter } from 'react-router-dom';

describe('Header', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <BrowserRouter>
                <Header />
            </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
