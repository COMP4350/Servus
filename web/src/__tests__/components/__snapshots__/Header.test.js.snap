// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
    position="static"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="makeStyles-title-2"
      >
        <img
          class="makeStyles-logo-3"
          height="50"
          src="servus_text.svg"
        />
      </div>
      <div
        class="makeStyles-headerOptions-6"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-menuButton-4"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4"
            >
              Appointments
            </h4>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-menuButton-4"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4"
            >
              Login
            </h4>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
